modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -621,6 +621,8 @@ Object encodeLiteral(Object literal, Class<?> targetType) {
             }
             // by default, return string as is
             return literal;
+        } else if (literal instanceof Number) {
+            return literal;
```

Comment:
I do not know what kinds of Numbers mongodb can hold, are we sure this won't trip up on BigDecimal or Byte? both of which are considered numbers.